JsonParser.cpp

```diff
@@ -0,0 +1,178 @@
+#include "JsonParser.h"
+#include <vector>
+#include <sstream>
+
+JsonParser::JsonParser(std::string stringInput) {
+    if (stringInput.find(".json") != std::string::npos) {
+        std::ifstream file(stringInput);
+        std::list<std::string> lines;
+        std::string line;
+        while (std::getline(file, line)) {
+            lines.push_back(line);
+        }
+        parseJson(lines);
+    } else {
+        std::stringstream ss(stringInput);
+        std::string temp = "";
+        std::list<std::string> lines;
+        while(std::getline(ss, temp)) {
+            lines.push_back(temp);
+        }
+        parseJson(lines);
+    }
+}
+
+JsonParser::JsonParser(std::istream& stream) {
+    std::list<std::string> lines;
+    std::string line;
+    while (std::getline(stream, line)) {
+        lines.push_back(line);
+    }
+    parseJson(lines);
+}
+
+
+double JsonParser::getDouble(std::string key) const {
+    return std::stod(data.at(key));
+}
+
+std::map<std::string, std::string> JsonParser::getData() const { 
+    return data; 
+}
+
+std::string JsonParser::getString(std::string key) const {
+    return data.at(key);
+}
+
+void JsonParser::parseJson(std::list<std::string>& list) {
+    bool isKey = false;
+    std::string line;
+    std::list<std::string> lines = list;
+    std::list<std::string> keyValues;
+    validateJson(lines);
+    lines.pop_back();
+    lines.pop_front();
+    
+    for (std::string line : lines) {
+        std::string trimmedLine = "";
+        for (char c : line) {
+            if (c == ' ' && isKey) {
+                trimmedLine += c;
+            } else if (c == '"' && !isKey) {
+                isKey = true;
+            } else if (c == '"' && isKey) {
+                isKey = false;
+            } else if (c == ' ' && !isKey) {
+                continue;
+            } else if (c == ',') {
+                continue;
+            } else {
+                trimmedLine += c;
+            }
+        }
+        keyValues.push_back(trimmedLine);
+    }
+    
+    lines.clear();
+    
+    for (std::string line : keyValues) {
+        int index = line.find(':');
+        std::string key = line.substr(0,index);
+        std::string value = line.substr(index + 1);
+        
+        data.insert(std::make_pair(key, value));
+    }
+
+    keyValues.clear();
+}
+
+bool JsonParser::validateJson(std::list<std::string> lines) {
+    std::vector<std::string> linesVector;
+
+    for (auto line : lines) { 
+        linesVector.push_back(line);
+    }
+    
+    for (int i = 0; i < linesVector.size(); i++) {
+        std::string tempString = "";
+        for (char iter : linesVector[i]) {
+            if (iter != ' ') {
+                tempString += iter;
+            }
+        }
+        linesVector[i] = tempString;
+    }
```

Comment:
Ha valakinek a neve tartalmaz szóközt, azt is ki fogjuk törölni? Igazából más fehér karakterek is előfordulhatnak, pl. tab, soremelés...

Reply:
Módosítottam a függvényt, akármennyi white karakter kerülhet bele.